### PR TITLE
Updated document to clarify Clip Behaviour

### PR DESCRIPTION
**Pull request by Neutrino2711**
Revised comments to clarify that clipping of child depends on clipBehavior of the parent widget, typically scrollable widgets that default to Clips.hard.
Noted that decoration features such as shadows , which render outside the widget boundary, may lead to undesirable effects.
Recommended using Clips.none in scenarios where shadow effects are used to avoid clipping issues.
Fixing Issue #156819

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Review by nate-thegrate — COMMENTED** [review]
Hi @Neutrino2711, thanks for contributing!

It looks like https://github.com/flutter/flutter/issues/156819 is asking for an example that turns off the parent ScrollView's clipping, and I agree that including an example would be ideal.

Would you be interested in creating an interactive DartPad code sample for this?

**Comment by Neutrino2711** [comment]
Yes @nate-thegrate i will be interested in doing so , i will start working on it

**Comment by Neutrino2711** [comment]
Hey @nate-thegrate I have a query , as solution to #156819 is observable through resizing of browser window and as DartPad can't directly control the size of window so should I simulate resizing of window or I should just simply add the correct example which wont be observable.

**Comment by nate-thegrate** [comment]
<details open> <summary>DartPad does allow resizing the window:</summary>

resize dartpad

</details>

Changing the height isn't possible however. Perhaps there could be a slider to control the scroll view items' heights.

**Comment by Neutrino2711** [comment]
https://github.com/user-attachments/assets/36d83569-e255-41d8-ac88-355762d63b14

I created this please do review @nate-thegrate

**Comment by nate-thegrate** [comment]
That looks awesome!

Currently we have a [**decorated_sliver.0.dart**](https://github.com/Neutrino2711/flutter/blob/update-documentation/examples/api/lib/widgets/sliver/decorated_sliver.0.dart) file, which is referenced in the documentation:

https://github.com/flutter/flutter/blob/ec04707febc94bd9d6cfdecd6e6314f9c99c3538/packages/flutter/lib/src/widgets/decorated_sliver.dart#L31-L35

Would you be able to add a **decorated_sliver.1.dart** file with this example, and reference it here too?

**Comment by Neutrino2711** [comment]
Yes I will be doing that.

**Review by nate-thegrate — COMMENTED** [review]
Love this example, especially the color scheme you chose!

I have a lot of suggestions; most of which are whitespace-related.

<br>

One other thing that needs to be done: create a **decorated_sliver.1_test.dart** file based on [**decorated_sliver.0_test.dart**](https://github.com/flutter/flutter/blob/master/examples/api/test/widgets/sliver/decorated_sliver.0_test.dart), and add maybe 1 or 2 simple tests.

**Comment by Neutrino2711** [comment]
I have a query why decorated_sliver.0.dart is imported like this  "import 'package:flutter_api_samples/widgets/sliver/decorated_sliver.0.dart' as example;" can I import it like this 
"import 'package:flutter/examples/api/lib/widgets/sliver/decorated_sliver.0.dart"

File: packages/flutter/lib/src/widgets/decorated_sliver.dart
Comment (nate-thegrate) [inline packages/flutter/lib/src/widgets/decorated_sliver.dart:40]:
```suggestion
/// This example demonstrates how the [CustomScrollView.clipBehavior]
/// impacts a decorated sliver's appearance.
///
/// The [Switch] determines whether clipping is enabled, and
/// the [Slider] adjusts the height of window.
```

File: packages/flutter/lib/src/widgets/decorated_sliver.dart
Comment (nate-thegrate) [inline packages/flutter/lib/src/widgets/decorated_sliver.dart:48]:
```suggestion
/// This widget does not apply any additional clipping to its [child].
/// To clip a child based on the [Decoration]'s shape, consider using
/// a [ClipPath] widget.
```

([relevant style guideline](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#use-correct-grammar))

File: packages/flutter/lib/src/widgets/decorated_sliver.dart
Comment (nate-thegrate) [inline packages/flutter/lib/src/widgets/decorated_sliver.dart:29]: I believe "the child is not clipped" is intended to mean "the child is not clipped by this `DecoratedSliver` widget", but I agree that the wording has some room for improvement 👍

File: examples/api/lib/widgets/sliver/decorated_sliver.1.dart
Comment (nate-thegrate) [inline examples/api/lib/widgets/sliver/decorated_sliver.1.dart:8]: Wrapping a class name with [square brackets] creates a link to that class, both in the IDE and on the [docs website](https://main-api.flutter.dev/), but it needs to be structured as a doc comment for this to work.

```suggestion
/// Flutter code example for [DecoratedSliver]
/// with clipping turned off in a parent [CustomScrollView].
```

File: examples/api/lib/widgets/sliver/decorated_sliver.1.dart
Comment (nate-thegrate) [inline examples/api/lib/widgets/sliver/decorated_sliver.1.dart:80]:
```suggestion
                padding: const EdgeInsets.all(24.0),
```

Just to make it look nicer :)

**Review by nate-thegrate — COMMENTED** [review]
Thanks for the update!

I believe this should be the last round of suggestions, though the autoformatter might make some of the whitespace changes difficult to perform.

You could get around this by committing via the GitHub site:
- head over to the [files tab](https://github.com/flutter/flutter/pull/157719/files)
- click **Add suggestion to batch** for all of them
- then there will be a button to commit the batch at the top of the screen

**Comment by Neutrino2711** [comment]
@nate-thegrate Thank you for your guidance.

**Review by nate-thegrate — APPROVED** [review]
LGTM, thanks for the contribution!

There were just a few analysis problems I missed previously, so go ahead and apply the suggestions when you get a chance. Then we'll be able to merge once the PR has 2 approvals.

<br>

There's a [wiki page](https://github.com/flutter/flutter/blob/master/docs/contributing/Setting-up-the-Framework-development-environment.md) that goes over how to set up your IDE so these linter rules are active while you're editing the file; feel free to check it out when you have a chance!

File: examples/api/lib/widgets/sliver/decorated_sliver.1.dart
Comment (nate-thegrate) [inline examples/api/lib/widgets/sliver/decorated_sliver.1.dart:159]: (oh, also 1 extra line here)

```suggestion
```

**Comment by nate-thegrate** [comment]
> I have a query why decorated_sliver.0.dart is imported like this "import 'package:flutter_api_samples/widgets/sliver/decorated_sliver.0.dart' as example;" in decorated_sliver.0_test.dart file can I import it like this "import 'package:flutter/examples/api/lib/widgets/sliver/decorated_sliver.0.dart"

Oh, forgot to answer your question, sorry! Specifying a namespace this way makes it easy to tell which classes are from the example and which ones aren't, and it's a convention followed consistently throughout example test files.

**Comment by Neutrino2711** [comment]
@nate-thegrate one last help how can I ask for review or how does it work I am new to this.

**Comment by nate-thegrate** [comment]
Sure, no prob!

The [Tree Hygiene](https://github.com/nate-thegrate/flutter/blob/fix-checks-wiki-page/docs/contributing/Tree-hygiene.md#who) wiki page has some info on this:

> PRs are assigned reviewers weekly. The precise process varies by team but tends to be combined with issue [triage](https://github.com/nate-thegrate/flutter/blob/fix-checks-wiki-page/docs/triage/README.md).

<br>

The next `team-framework` triage is tomorrow, so we'll figure out a second reviewer then :)

File: examples/api/lib/widgets/sliver/decorated_sliver.1.dart
Comment (chunhtai) [inline examples/api/lib/widgets/sliver/decorated_sliver.1.dart:37]: super tiny nits. lets keep the styling consistent
```suggestion
  bool _isClipped = false;
```

**Review by chunhtai — APPROVED** [review]
LGTM, just one nitpicking

File: examples/api/lib/widgets/sliver/decorated_sliver.1.dart
Comment (chunhtai) [inline examples/api/lib/widgets/sliver/decorated_sliver.1.dart:53]: All reference will need to be changed too

**Pull request merged**
